gog/graph.go

```diff
@@ -47,6 +47,16 @@ type Grapher struct {
 
 	seenDocObjs map[types.Object]struct{}
 	seenDocKeys map[string]struct{}
+
+	// pkgDefs accumulates the defs from the package being graphed currently in Graph().
+	// If the package is graphed successfully, the defs are added to Output.
+	pkgDefs []*Def
```

Comment:
`gog` is actually not operating at individual source unit level. It loads all packages and dependencies of the 'program' (ie repo) [here](https://github.com/sourcegraph/srclib-go/blob/master/gog/graph.go#L70-L74) and then `g.Graph` is called for each package on the same `gog` object [here](https://github.com/sourcegraph/srclib-go/blob/master/graph.go#L369-L373). This is the crux of all the efficiency savings from GraphMultiUnitsRule, because earlier `gog` would load all dependencies repeatedly for all packages of the repo.